packages/x-data-grid/src/components/virtualization/GridVirtualScrollerContent.tsx

```diff
@@ -28,10 +30,15 @@ const GridVirtualScrollerContent = React.forwardRef<
   HTMLDivElement,
   React.HTMLAttributes<HTMLDivElement> & { sx?: SxProps<Theme> }
 >(function GridVirtualScrollerContent(props, ref) {
+  const apiRef = useGridApiContext();
   const rootProps = useGridRootProps();
   const overflowedContent = !rootProps.autoHeight && props.style?.minHeight === 'auto';
   const classes = useUtilityClasses(rootProps, overflowedContent);
 
+  useEnhancedEffect(() => {
+    apiRef.current.publishEvent('virtualScrollerContentSizeChange');
+  }, [apiRef, props.style]);
+
```

Comment:
I completely agree with you there, and just finishing a commit on that note. Wasn't completely obvious to me initially how I could do it in a clean way, as it needs to run before paint to avoid triggering reflows – so at useLayoutEffect of the ScrollerContent component.

Reply:
Fixed with a callbackRef. Also made it more complete for different use cases. E.g. when columns are not immediately available, and if columns and rows become available at different times.

`ColumnVirtualizationGrid.tsx` in the docs turned out be a good test case.